vcpkg/produce/test-registry-ports-gha.md

```diff
@@ -0,0 +1,297 @@
+[asset cache](../users/assetcaching.md) and [binary
+cache](../users/binarycaching.md) features.
+
+## Set up a workflow to test your registry's ports
+
+After you have set up a binary cache and asset cache for your CI environment,
+the next step is to set up a workflow to test all your registry's ports. You can
+decide whether this workflow runs on a schedule or if it is triggered by new
+commits or Pull Requests.
+
+The main vcpkg registry uses the `vcpkg ci` command, which has been tailored to
+the vcpkg project's needs and is not intended to remain stable or be used by
+consumers of vcpkg. As such, it is not suited to use for testing your own vcpkg
+registries. We recommend to follow the steps outlined in this article instead.
+
+### Use a manifest file to include all your ports
+
+Instead of using the `vcpkg ci` command, we suggest using a manifest file to
+create a build that depends on all the packages in your registry.
+
+The following example, creates a manifest file to test all the ports in a
+hypothetical vcpkg registry. Replace the list of dependencies to include all the
+ports in your registry and place it in the root of your repository.
+
+`vcpkg.json`
+
+```json
+{
+  "dependencies": [
+    "beicode",
+    "beison"
+  ]
+}
+```
+
+Your own ports may have dependencies on the main vcpkg registry or other
+third-party registries, in which case, you need to add those registries in a
+`vcpkg-configuration.json` file. While vcpkg can resolve packages from the main
+registry without additional configuration, we strongly recommend that you explicitly
+add it to the registries list for version control purposes.
+This ensures that you have control of the set of underlying port
+versions. Check out the [`vcpkg x-update-baseline`
+command](../commands/update-baseline.md) to help managing the baseline of your
+registries.
+
+`vcpkg-configuration.json`
+
+```json
+{
+  "registries": [
+    {
+      "kind": "git",
+      "repository": "https://github.com/Microsoft/vcpkg",
+      "baseline": "42bb0d9e8d4cf33485afb9ee2229150f79f61a1f",
+      "packages": "*"
+    }
+  ]
+}
+```
+
+Read the [`vcpkg.json`](../reference/vcpkg-json.md) and
+[`vcpkg-configuration.json`](../reference/vcpkg-configuration-json.md) reference
+articles to learn more. And the [manifest mode
+documentation](../consume/manifest-mode.md) to learn how these work together.
+
+### Acquire vcpkg in your GitHub Actions workflow
+
+Next, you need to acquire vcpkg to use it in your workflow. Add the following
+steps to install vcpkg.
+
+```yml
+steps:
+- uses: actions/checkout@v4
+  with:
+    repository: "https://github.com/Microsoft/vcpkg"
+    fetch-depth: 0
+
+- name: Bootstrap vcpkg
+  run: ${{ github.worksapce }}/vcpkg/bootstrap.sh
+  shell: bash
+```
+
+Make sure that you set `fetch-depth` to 0 when checking out the vcpkg repository
+since you need the full repository's commit history for port versioning to work.
```

Comment:
Is the "full" history really required? (I understand that this instruction avoids problems, but it is more like a "unless you know what you are doing".)

Reply:
A shallow clone is fine since we're adding the vcpkg registry as a Git registry in `vcpkg-configuration.json`. So maybe you're right and we should remove that.